libraries/standard/mqtt/src/iot_mqtt_api.c

```diff
@@ -1765,37 +1765,43 @@ IotMqttError_t IotMqtt_PublishSync( IotMqttConnection_t mqttConnection,
                                     uint32_t timeoutMs )
 {
     IotMqttError_t status = IOT_MQTT_STATUS_PENDING;
-    IotMqttOperation_t publishOperation = IOT_MQTT_OPERATION_INITIALIZER,
-                       * pPublishOperation = NULL;
-    /* Set only the "serial" flag. */
-    uint32_t syncFlags = MQTT_INTERNAL_FLAG_BLOCK_ON_SEND;
-
-    /* Flags are currently ignored. */
-    ( void ) flags;
-
-    /* Set the waitable flag and reference for QoS 1 PUBLISH. */
-    if( pPublishInfo->qos == IOT_MQTT_QOS_1 )
+    if( pPublishInfo == NULL )
     {
-        syncFlags |= IOT_MQTT_FLAG_WAITABLE;
-        pPublishOperation = &publishOperation;
+        status = IOT_MQTT_BAD_PARAMETER;
     }
+    else
+    {
+        IotMqttOperation_t publishOperation = IOT_MQTT_OPERATION_INITIALIZER,
```

Comment:
All variables must be declared at the start of the function.

Reply:
Done.